lib/util/url.ts

```diff
@@ -1,3 +1,20 @@
+import urlJoin from 'url-join';
+
 export function ensureTrailingSlash(url: string): string {
   return url.replace(/\/?$/, '/');
 }
+
+export function resolveBaseUrl(baseUrl: string, input: string | URL): string {
+  const inputString = input.toString();
+
+  let host;
+  let pathname;
+  try {
+    ({ host, pathname } = new URL(inputString));
+  } catch (e) {
+    pathname = inputString;
+  }
+
+  const result = host ? inputString : urlJoin(baseUrl, pathname || '');
+  return result.replace(/\/+$/, '');
```

Comment:
Are we certain that we always want to strip the trailing slash?

Reply:
Let's preserve, except strings like these `/foo/?bar=baz`